calliope/backend/pyomo/model.py

```diff
@@ -74,42 +75,23 @@ def generate_model(model_data):
     )
 
     for k, v in model_data_dict["data"].items():
-        if k in backend_model.__calliope_defaults.keys():
-            setattr(
-                backend_model,
-                k,
-                po.Param(
-                    *[getattr(backend_model, i) for i in model_data_dict["dims"][k]],
-                    initialize=v,
-                    mutable=True,
-                    default=backend_model.__calliope_defaults[k],
-                ),
-            )
+        _kwargs = {
+            "initialize": v,
+            "mutable": True,
+            "within": getattr(po, get_domain(model_data[k])),
+        }
+        if not pd.isnull(backend_model.__calliope_defaults.get(k, None)):
```

Comment:
Is `model.__calliope_default` a dictionary (or mapping)?  Then you could just do:
```suggestion
        if k in backend_model.__calliope_defaults:
```

Reply:
It's still possible for k to be in the dictionary and have a value of None, this is aimed at catching *both* those keys not in the dictionary and those in the dictionary but with a value of 'None' or 'NaN'

Reply:
Hmm, whether a config should have a `None` default is a debate for another time.  At least after I fix the `dataconfig` issues :-p